pygatt/bled112_backend.py

```diff
@@ -211,9 +216,9 @@ def bond(self):
         self._lib.send_command(self._ser, cmd)
 
         # Wait for response
-        self._loglock.release()  # don't hold loglock while waiting
-        self._wait_for_cmd_response()
-        self._loglock.acquire()
+        self._process_packets_until(
+            [self._lib.PacketType.ble_rsp_sm_set_bondable_mode,
+             self._lib.PacketType.ble_evt_connection_disconnected])
```

Comment:
What are the cases where we don't want to handle a disconnected event?  I'm curious if we can make it a special case and handle it in one place (and if that makes sense)?